### PR TITLE
Merge relationship editor changes from the test branch into beta

### PR DESCRIPTION
**Pull request by mwiencek**
# Problem

In order to iterate on feedback from MBS-11847 and other small relationship editor tickets more quickly in preparation for the next beta update, I've been updating the test branch with changes and deploying test.mb so that people can try out the changes.  Now these changes need to be reviewed and synced to the beta branch.

# Solution

Merge the test branch into beta.

This fixes several tickets:
 * MBS-12834
 * MBS-12849
 * MBS-12850
 * MBS-12830

### CODE REVIEW

**Comment by reosarevok** [comment]
@brainzbot, retest this please

File: root/static/scripts/edit/components/HelpIcon.js [hunk 1]
```diff
@@ -15,32 +15,24 @@ type PropsT = {
...
-  const [hover, setHover] = React.useState(false);
-  return (
-    <div
-      style={{
-        position: 'relative',
-        display: 'inline-block',
-        marginLeft: '10px',
-      }}
-    >
+}: PropsT): React.MixedElement => (
```
Comment (reosarevok) [inline root/static/scripts/edit/components/HelpIcon.js:26]: This is just `Tooltip`? Same for at least `BatchCreateWorksButtonPopover` which I cannot leave a comment on for reasons.

File: root/static/scripts/relationship-editor/components/RelationshipDialogContent.js [hunk 2]
```diff
@@ -526,23 +526,29 @@ const AttributesSection = (React.memo<AttributesSectionPropsT>(({
...
         <div className="heading-line" />
         <span className="heading-text">
           {l('Attributes')}
-          {' '}
-          {bracketed(
-            <a href="#" onClick={handleAttributesHelpClick}>
-              {l('help')}
-            </a>,
-          )}
+          {hasAttributes ? (
```
Comment (reosarevok) [inline root/static/scripts/relationship-editor/components/RelationshipDialogContent.js:550]: This should probably be reverted if we're adding back the dates help (which we should).

**Review by reosarevok — CHANGES_REQUESTED** [review]
This generally seems good. Please do rescue the help text for dates though (and I guess drop the related commit here since there'll always be something with docs then if the attributes section exists).

File: root/static/scripts/relationship-editor/components/DialogPreview.js [hunk 3]
```diff
@@ -82,7 +82,7 @@ const DialogPreview = (React.memo<PropsT>(({
   source,
   newRelationship,
   oldRelationship,
-}: PropsT): React.Element<'fieldset'> => {
+}: PropsT): React.MixedElement => {
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply root/static/scripts/relationship-editor/components/DialogPreview.js:85]: I'd probably only use Fragment if the users of the component demanded a fragment somehow (but that's not the case, so it doesn't add any type safety to be overly specific).

**Comment by kellnerd** [comment]
[MBS-12834](https://tickets.metabrainz.org/browse/MBS-12834) still doesn't appear to be solved for me, the relationship type menu still appears when I'm editing an existing relationship on test.

**Comment by mwiencek** [comment]
Thanks, it's working in Chromium but I hadn't tested it in Firefox where it appears to be broken still.  I'll take another look at it.

File: root/static/scripts/relationship-editor/components/RelationshipTargetTypeGroups.js [hunk 4]
```diff
@@ -87,7 +97,14 @@ const RelationshipTargetTypeGroups = (React.memo<PropsT>(({
...
         {sections}
         <tr>
-          <td colSpan="2">
+          <td
+            className="add-relationship"
+            /*
+             * Without any relationship sections, this cell stretches
+             * which causes the button alignment to change sides.
+             */
+            style={sections.length ? null : {width: '160px'}}
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply root/static/scripts/relationship-editor/components/RelationshipTargetTypeGroups.js]: It seems to work if I just set `width: 160px;` unconditionally in the .less file.

**Comment by mwiencek** [comment]
> This generally seems good (and I've played with it in test quite a bit and was happy with it). Please do rescue the help text for dates though (and I guess drop the related commit here since there'll always be something with docs then if the attributes section exists).
> 
> Ideally we can also fix the issue with the double keyboard vs mouse focus on dropdowns before merging (or in a second beta PR after merge).

The date help text has now been resurrected and the double-menu-highlighting has been fixed.

**Comment by reosarevok** [comment]
5c9ff49607524b55b3984f4e46f3f4bb691c554d is still there though hiding stuff if there's only dates?

**Comment by mwiencek** [comment]
> https://github.com/metabrainz/musicbrainz-server/commit/5c9ff49607524b55b3984f4e46f3f4bb691c554d is still there though hiding stuff if there's only dates? (and the dates help is not showing for me in test.mb)

Yeah, I must've of screwed up a rebase since the new commit is missing.  Recovered it from the reflog.

**Review by reosarevok — APPROVED** [review]
Tested the latest changes locally and it all feels good to me, and the code also seems sensible. Ship it to beta IMO.

**Pull request merged**